lib/modules/manager/npm/types.ts

```diff
@@ -0,0 +1,67 @@
+interface LockFilePackage {
+  name?: string;
+  version?: string;
+  resolved?: string;
+  integrity?: string;
+  link?: boolean;
+  dev?: boolean;
+  optional?: boolean;
+  devOptional?: boolean;
+  inBundle?: boolean;
+  hasInstallScript?: boolean;
+  hasShrinkwrap?: boolean;
+  bin?: string | Record<string, string>;
+  license?: string;
+  engines?: Record<string, string>;
+  dependencies?: Record<string, string>;
+  optionalDependencies?: Record<string, string>;
+}
+
+interface LockFileDependency {
+  version: string;
+  integrity: string;
+  resolved: string;
+  bundled?: boolean;
+  dev?: boolean;
+  optional?: boolean;
+  requires?: Record<string, string>;
+  dependencies?: Record<string, LockFileDependency>;
+}
+
+// https://docs.npmjs.com/cli/v6/configuring-npm/package-lock-json#file-format
+interface LockFile1 {
+  name: string;
+  version: string;
```

Comment:
Can we extract a base interface which shares same properties for all lockfile versions to not need to retype everything again and again?